sdk/servicebus/service-bus/test/batchReceiver.spec.ts

```diff
@@ -1190,4 +1191,245 @@ describe("Batching - disconnects", function(): void {
     settledMessageCount.should.equal(2, "Unexpected number of settled messages.");
     refreshConnectionCalled.should.be.greaterThan(0, "refreshConnection was not called.");
   });
+
+  it("returns messages if drain is in progress (receiveAndDelete)", async function(): Promise<
+    void
+  > {
+    // Create the sender and receiver.
+    sbClient = getServiceBusClient();
+    const { receiverClient, senderClient } = await getSenderReceiverClients(
+      sbClient,
+      TestClientType.UnpartitionedQueue,
+      TestClientType.UnpartitionedQueue
+    );
+    const receiver = receiverClient.createReceiver(ReceiveMode.receiveAndDelete);
+    const sender = senderClient.createSender();
+
+    // The first time `receiveMessages` is called the receiver link is created.
+    // The `receiver_drained` handler is only added after the link is created,
+    // which is a non-blocking task.
+    await receiver.receiveMessages(1, 1);
+
+    if (!receiver["_context"].batchingReceiver!.isOpen()) {
```

Comment:
This isn't needed, is it? ie, the previous line should have thrown an error if init() failed outright?

Reply:
cc @chradek